### ISSUE TITLE
invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by ucwong** [anacrolix/dht#51]
```
^[[0mpanic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x10de6c7]
/vendor/github.com/anacrolix/dht/v2/bep44/store.go:54 +0x27
```

**Comment by anacrolix** [comment]
Could you provide more of the stack trace?

**Comment by anacrolix** [comment]
@ajnavarro possibly the store isn't initialized for default configs and users are trying to handle BEP44 messages?

**Comment by ucwong** [comment]
```
^[[0mpanic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x10de6c7]

goroutine 13 [running]:
github.com/CortexFoundation/CortexTheseus/vendor/github.com/anacrolix/dht/v2/bep44.(*Wrapper).Get(0xc0003f8318, {0x74, 0x5b, 0x78, 0xa, 0xc3, 0xad, 0xed, 0x8a, 0xb4, ...})
        /root/CortexTheseus/build/_workspace/src/github.com/CortexFoundation/CortexTheseus/vendor/github.com/anacrolix/dht/v2/bep44/store.go:54 +0x27
github.com/CortexFoundation/CortexTheseus/vendor/github.com/anacrolix/dht/v2.(*Server).handleQuery(0xc0000b4000, {0x1dee258, 0xc02b8f6bc0}, {{0xc028915867, 0x3}, 0xc031c045a0, {0xc02891586c, 0x4}, {0xc028915871, 0x1}, ...})
        /root/CortexTheseus/build/_workspace/src/github.com/CortexFoundation/CortexTheseus/vendor/github.com/anacrolix/dht/v2/server.go:577 +0x405
github.com/CortexFoundation/CortexTheseus/vendor/github.com/anacrolix/dht/v2.(*Server).processPacket(0xc0000b4000, {0xc0005f6000, 0x58, 0x10000}, {0x1dee258, 0xc02b8f6bc0})
        /root/CortexTheseus/build/_workspace/src/github.com/CortexFoundation/CortexTheseus/vendor/github.com/anacrolix/dht/v2/server.go:298 +0x310
github.com/CortexFoundation/CortexTheseus/vendor/github.com/anacrolix/dht/v2.(*Server).serve(0xc0000b4000)
        /root/CortexTheseus/build/_workspace/src/github.com/CortexFoundation/CortexTheseus/vendor/github.com/anacrolix/dht/v2/server.go:344 +0x1b7
github.com/CortexFoundation/CortexTheseus/vendor/github.com/anacrolix/dht/v2.(*Server).serveUntilClosed(0xc0000b4000)
        /root/CortexTheseus/build/_workspace/src/github.com/CortexFoundation/CortexTheseus/vendor/github.com/anacrolix/dht/v2/server.go:227 +0x33
created by github.com/CortexFoundation/CortexTheseus/vendor/github.com/anacrolix/dht/v2.NewServer
        /root/CortexTheseus/build/_workspace/src/github.com/CortexFoundation/CortexTheseus/vendor/github.com/anacrolix/dht/v2/server.go:222 +0x4b0
```

**Comment by ucwong** [comment]
> @ajnavarro possibly the store isn't initialized for default configs and users are trying to handle BEP44 messages?

https://github.com/anacrolix/dht/blob/master/server.go#L577

**Comment by ajnavarro** [comment]
> @ajnavarro possibly the store isn't initialized for default configs and users are trying to handle BEP44 messages?

Yes, I think that is the problem. I created a PR addressing the problem.

**Comment by anacrolix** [comment]
I didn't realise Store was set by default https://github.com/anacrolix/dht/blob/0f19ece30d00804039e55c2f57d5e56ac359c58e/server.go#L162. @ucwong are you providing your own server config? In that case, since the old behaviour was not to handle get and put messages, we should probably default to not handling those if a store is not provided (instead of creating a default one). What do you think?

**Comment by ucwong** [comment]
> I didn't realise Store was set by default
> 
> https://github.com/anacrolix/dht/blob/0f19ece30d00804039e55c2f57d5e56ac359c58e/server.go#L162
> 
> . @ucwong are you providing your own server config? In that case, since the old behaviour was not to handle get and put messages, we should probably default to not handling those if a store is not provided (instead of creating a default one). What do you think?

https://github.com/anacrolix/dht/blob/0f19ece30d00804039e55c2f57d5e56ac359c58e/server.go#L180
I think ```c.store``` will be nil with customized config file

https://github.com/anacrolix/torrent/blob/master/client.go#L381-L398
I am not using dht directly but ```c.store``` seems not to be initialized in this case

**Comment by anacrolix** [comment]
I've decided to have this enabled by default. More nodes supporting BEP44 the better.

**Comment by anacrolix** [comment]
Fixed in v2.12.0.
